src/pages/index.js

```diff
@@ -2,101 +2,232 @@ import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
 import '../pages/index.css';
 import PopupWithImage from "../components/PopupWithImage.js";
+import PopupWithDelete from "../components/PopupWithDelete.js"
 import PopupWithForm from "../components/PopupWithForm.js";
 import Section from "../components/Section.js";
 import UserInfo from "../components/UserInfo.js";
+import Api from "../components/Api.js";
+import Popup from "../components/Popup.js";
 import {
   initialCards,
   cardListEl,
   profileEditBtn,
   profileEditForm,
   addCardForm,
+  addCardModal,
+  avatarButton,
+  addCardButton,
   addNewCardButton,
   profileTitleInput,
   profileDescriptionInput,
+  avatarForm,
   settings,
-} from "../utils/components.js";
+} from "../utils/constants.js";
 import "./index.css";
 
 //const card = new Card(cardData, '#card-template');
 //card._getView();
 
-//Class Instances
-const previewImageModal = new PopupWithImage("#preview-image-modal");
-previewImageModal.setEventListeners();
-
-const addCardModal = new PopupWithForm("#add-card-modal", handleAddCardSubmit);
-addCardModal.setEventListeners();
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "41597df2-eb93-4b41-8bf1-f4a542a9ea54",
+    "Content-Type": "application/json"
+  }
+});
 
-const profileEditModal = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileEditSubmit
-);
-profileEditModal.setEventListeners();
+//Class Instances
+const previewImageModal = new PopupWithImage({ popupSelector: "#preview-image-modal" });
+const popupDelete = new PopupWithDelete({ popupSelector: "#delete-card-modal" });
+
+const userInfo = new UserInfo(".profile__title", ".profile__description", ".profile__image");
+
+let section;
+
+api.getInitialCards().then((initialCards) => {
+  section = new Section(
+    {
+      items: initialCards,
+      renderer: (data) => {
+        const cardElement = new Card(
+          data,
+          "#card-template",
+          handleImageClick,
+          handleDelete,
+          handleLikeCard
+        );
+        section.addItems(cardElement.getView());
+      },
+    },
+    ".cards__list"
+  );
+  section.renderItems();
+});
```

Comment:
Needs correction: don't forget to handle the api errors, add a `catch` block (fix that everywhere)

Reply:
I added catch blocks to all apis